src/Umbraco.Core/Composing/Composition.cs

```diff
@@ -133,7 +133,11 @@ public IFactory CreateFactory()
 
             Configs.RegisterWith(_register);
 
-            return _register.CreateFactory();
+            IFactory factory = null;
+            // ReSharper disable once AccessToModifiedClosure -- on purpose
+            _register.Register(_ => factory, Lifetime.Singleton);
```

Comment:
@zpqrtbnk why not put this line underneath the call to `factory = _register.CreateFactory();` ? Not sure i understand the need to declare `IFactory factory = null;` why not just:

```cs
var factory = _register.CreateFactory();
 _register.Register(_ => factory, Lifetime.Singleton);
return factory;
```

Reply:
Because, for some containers, you cannot register anything after the factory has been created.

And so, we need to register first (capturing a reference to a still-null factory), then create the factory and assign it to the reference.

Making sense?

Reply:
ok cool, just wanted to check